code/obj/nuclearreactor/nuclearreactor.dm

```diff
@@ -839,6 +839,20 @@
 		return TRUE
 
 	tick(var/obj/projectile/P)
+		var/turf/simulated/T = get_turf(P)
+		if (istype(T, /turf/space/fluid)) //can't really calculate this, just make it do something at least
+			P.power -= 10
+		else if (issimulatedturf(T))
+			var/obj/fluid/fluid = T.active_liquid
+			if (fluid)
+				var/viscosity = max(fluid.group.avg_viscosity, 10) //fluid group viscocity seems to be either 1 or on a scale between 10-20, why
+				P.power -= viscosity * fluid.my_depth_level / 2
+				fluid.group.reagents.add_reagent("radium", P.power / 10)
+
+			if (T.air?.carbon_dioxide)
+				P.power -= min(T.air.carbon_dioxide / 7, 40)
+				T.air.radgas += 1
```

Comment:
I feel like this should also remove part of the CO2.

But more importantly as written this does not work when group processing is on I believe. You will probably want to check `T.parent?.group_processing` and if it is true then do this all for `T.parent.air` except divided by `T.parent.air.group_multiplier`.

Reply:
ohh right, I was reading the reactor component code which doesn't use group processing

Reply:
If we're doing gas checks anyway, do you maybe want to just take the neutron interaction code from the gas cooling reactor component and generalise it to this? Then we can have neutrons interacting with plasma in the air and making fallout and all that jazz.